ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -68,9 +74,13 @@ const PermitSimulation: React.FC<{
         <ConfirmInfoRowText text={t('permitSimulationDetailInfo')} />
       </ConfirmInfoRow>
       <ConfirmInfoRow label={t('spendingCap')}>
-        <Box style={{ marginLeft: 'auto' }}>
+        <Box style={{ marginLeft: 'auto', maxWidth: '100%' }}>
           <Box display={Display.Flex}>
-            <Box display={Display.Inline} marginInlineEnd={1}>
+            <Box
+              display={Display.Inline}
+              marginInlineEnd={1}
+              minWidth={BlockSize.Zero}
```

Comment:
The more complexity we add here, I worry we diverge from transaction simulations and it seems to add more value to the prospect of a centralised component such as `SimulationNumber` or `ConfirmLargeNumber` that encapsulates all the math, logic, and rendering for these values and tooltips?

Reply:
I agree with you that it be better to separate concerns and create centralised components. I have been thinking about this too as I've been working on it. 

Initially, I was only working on adding ellipsis to unblock v12.1. Working on this, I uncovered the BigNumber error and realized it also needs to be fixed for v12.1. I think we should keep cherry-pick changes at a minimum.

We are also working on introducing many number displays on Signature Redesign which is when I was thinking is a good time to abstract the logic into a reusable component. I don't mind if there is repeated logic in two places before we create a reusable component. A couple of the few PRs that will introduce more values displayed in rows:
- https://github.com/MetaMask/MetaMask-planning/issues/2644
- https://github.com/MetaMask/MetaMask-planning/issues/2666

Thanks for bringing it up here for discussion!